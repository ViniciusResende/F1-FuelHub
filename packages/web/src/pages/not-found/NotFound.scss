@use '../../styles/variables' as *;

.wrapper {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1.5rem;
  padding: 2rem;
  text-align: center;
  background: $backgroundColor;
  color: $fontColor;

  button {
    width: 100%;
    max-width: 300px;
  }
}

.illustration {
  width: min(320px, 60vw);
  height: auto;
}

.title {
  font-size: 2.5rem;
  font-weight: 700;
}

.text {
  font-size: 1.125rem;
  opacity: 0.8;
}
