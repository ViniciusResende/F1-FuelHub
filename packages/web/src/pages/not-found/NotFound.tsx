/** Next Js */
import { useRouter } from 'next/router';

/** Components */
import Button from '../../components/Button';

/** Assets */
import { CarRepairIllustration } from '../../assets/svg/illustrations';

/** Styles */
import './NotFound.scss';

function NotFound() {
  const router = useRouter();

  const goHome = () => router.push('/');

  return (
    <main className='wrapper'>
      <CarRepairIllustration className='illustration' />
      <h1 className='title'>404 â€“ Page Not Found</h1>
      <p className='text'>Looks like you took a wrong turn in the pit lane.</p>

      <Button onClick={goHome} modifier='default'>
        Back to Home
      </Button>
    </main>
  );
}

/** Exports */
export default NotFound;
