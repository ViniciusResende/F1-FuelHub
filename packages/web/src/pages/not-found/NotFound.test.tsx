/** Jest */
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

/** Components */
import NotFound from './NotFound';

/** Mocks */
const push = jest.fn();

jest.mock('next/router', () => ({
  useRouter: () => ({ push }),
}));

jest.mock('../../assets/svg/illustrations', () => ({
  CarRepairIllustration: (props: any) => (
    <svg data-testid='illustration' {...props} />
  ),
}));

/** Tests */

describe('NotFound page', () => {
  beforeEach(() => push.mockClear());

  it('renders correctly (snapshot)', () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('redirects to home when the button is clicked', async () => {
    const user = userEvent.setup();
    render(<NotFound />);

    const button = screen.getByRole('button', { name: /back to home/i });
    await user.click(button);

    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith('/');
  });
});
