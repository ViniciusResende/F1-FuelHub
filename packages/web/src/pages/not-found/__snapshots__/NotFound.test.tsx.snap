// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page renders correctly (snapshot) 1`] = `
<DocumentFragment>
  <main
    class="wrapper"
  >
    <svg
      class="illustration"
      data-testid="illustration"
    />
    <h1
      class="title"
    >
      404 â€“ Page Not Found
    </h1>
    <p
      class="text"
    >
      Looks like you took a wrong turn in the pit lane.
    </p>
    <button
      class="button-component default"
    >
      Back to Home
    </button>
  </main>
</DocumentFragment>
`;
